code/rounds/WaitingRound.cs

```diff
@@ -0,0 +1,71 @@
+using Sandbox;
+using System.Threading.Tasks;
+
+using TTTReborn.Player;
+
+namespace TTTReborn.Rounds
+{
+    public class WaitingRound : BaseRound
+    {
+        public override string RoundName => "Waiting";
+
+        public override void OnSecond()
+        {
+            if (Host.IsServer)
+            {
+                CheckMinimumPlayers();
+            }
+        }
+
+        public override void OnPlayerKilled(TTTPlayer player)
+        {
+            _ = StartRespawnTimer(player);
+
+            base.OnPlayerKilled(player);
+        }
+
+        public override void OnPlayerSpawn(TTTPlayer player)
+        {
+            if (Players.Contains(player))
+            {
+                return;
+            }
+
+            AddPlayer(player);
+
+            base.OnPlayerSpawn(player);
+        }
+
+        protected override void OnStart()
+        {
+            if (Host.IsServer)
+            {
+                foreach (Client client in Client.All)
+                {
+                    if (client.Pawn is TTTPlayer player)
+                    {
+                        player.Respawn();
+                    }
+                }
+            }
+        }
+
+        private async Task StartRespawnTimer(TTTPlayer player)
+        {
+            await Task.Delay(1000);
+
+            if (TTTReborn.Gamemode.Game.Instance.Round is WaitingRound)
```

Comment:
Might not even be needed, or could even use `this` instead. Would test but my IDE keeps fighting will me, hopefully https://github.com/TTTReborn/ttt-reborn/pull/8 will help standardize a lot of things.